src/nupic/engine/NetworkFactory.cpp

```diff
@@ -0,0 +1,202 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2013, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+#include <iostream>
+
+#include <nupic/engine/Input.hpp>
+#include <nupic/engine/Output.hpp> 
+#include <nupic/engine/Network.hpp>
+#include <nupic/engine/NetworkFactory.hpp>
+#include <nupic/engine/Region.hpp>
+#include <nupic/os/Env.hpp>
+#include <nupic/os/FStream.hpp>
+#include <nupic/os/OS.hpp>
+#include <nupic/os/Path.hpp>
+#include <nupic/utils/Log.hpp>
+#include <nupic/utils/StringUtils.hpp>
+
+namespace nupic
+{
+  Network* NetworkFactory::createNetwork(const std::string& path)
+  { 
+    if (!StringUtils::endsWith(path, ".yaml"))
```

Comment:
I don't think this is necessary. ".yml" is common and even if it is something else it is better to let it fail on parsing.